src/Services/SearchService.php

```diff
@@ -154,7 +154,7 @@ public function doSearch(HttpRequest $request, ExternalSearch $externalSearch)
     {
         $results = $this->search($request, $externalSearch);
 
-        if ($results->getResultsCount() == 0) {
+        if ($results->getResultsCount() == 0 && !$results->getLandingPage()) {
```

Comment:
This little thing took me ages to figure out.. Otherwise this would've been a 2min fix.

Reply:
As result count and landingpage is independent, how about adding the redirect to landingpage above this one? Landingpage has priority anyway.